includes/class-wc-payments-payment-request-button-handler.php

```diff
@@ -499,6 +499,11 @@ public function should_show_payment_request_button() {
 			return false;
 		}
 
+		// Order total doesn't matter for Pay for Order page. Thus, this page should always display payment buttons.
+		if ( $this->is_pay_for_order_page() ) {
```

Comment:
shouldn't we check if all the products are supported by the payment request button ( most of the conditions from `has_allowed_items_in_cart` method ) ?

Reply:
Hmmm, I think you are right! I'm unfamiliar with the logic and helpers here, but it looks like `$this->is_product_supported()` does that check right?

Reply:
yes. I think so.

Reply:
Moved the check a little down as discussed.

Reply:
I agree with @elazzabi, moving the check down LGTM.